src/commands/sbom/license.ts

```diff
@@ -0,0 +1,93 @@
+import {DependencyLicense} from './types'
+
+// Get the license from a string. If the license is valid, we return it. Otherwise, we return undefined
+const getLicenseFromString = (s: string): DependencyLicense | undefined => {
+  if (!s) {
+    return undefined
+  }
+
+  switch (s.toLowerCase()) {
+    case '0bsd':
+      return DependencyLicense.ZEROBSD
+    case 'apache-2.0':
+      return DependencyLicense.APACHE2
+    case 'bsd-2-clause':
+      return DependencyLicense.BSD2CLAUSE
+    case 'bsd-3-clause':
+      return DependencyLicense.BSD3CLAUSE
+    case 'bsl-1.0':
+      return DependencyLicense.BSL1
+    case 'isc':
+      return DependencyLicense.ISC
+    case 'mit':
+      return DependencyLicense.MIT
+    case 'zlib':
+      return DependencyLicense.ZLIB
+  }
+
+  console.debug(`license ${s} not recognized`)
+
+  return undefined
+}
+
+// Get all the licenses from a string. Sometimes, there are two licenses in one string
+// such as "MIT OR Apache-2.0". In this case, we return all the licenses in this condition.
+export const getLicensesFromString = (s: string): DependencyLicense[] => {
+  if (!s) {
+    return []
+  }
+  const licenses: DependencyLicense[] = []
+
+  if (s.indexOf('OR') !== -1) {
+    for (const lic of s.toLowerCase().split(' or ')) {
+      const l = getLicenseFromString(lic.replace(' ', ''))
```

Comment:
Why are you removing spaces? I don't see any test cases about that.

If it's to remove possible duplicate spaces like `aaa OR    bbb`, then you can use `.trim()`.

But if think there may be something like `Apache 2.0`, then removing spaces won't work with your switch case in `getLicenseFromString()`

Reply:
> There is never a space in a license

Ok 🙂 And since `.split(' or ')` includes the spaces in the delimiter, your values are `['MIT', 'Apache-2.0']` so I think you should be able to remove `.replace(' ', '')`

Reply:
The problem with `split` is that if I have `MIT OR  Apache-2.0`, it will not work. Some files have this, which is also why I remove the spaces